packages/@sfcc-core/core-graphql/src/graphql.ts

```diff
@@ -172,13 +172,48 @@ export type AppContext = PassportContext<User, AuthenParams> & {
     setSessionProperty: (prop: string, value?: string) => void;
 };
 
-export async function getUserFromContext(context: AppContext) {
+export async function getUserFromContext(context: AppContext, refresh = false) {
     let user = context.getUser();
-    const token = user ? user.token : '';
+    const token = user && !refresh ? user.token : '';
     if (!token) {
         const res = await context.authenticate('graphql-local', { token });
         context.login(res.user);
         user = res.user;
     }
     return user;
 }
+
+/**
+ * Return true if the current API authorizaion token has expired.
+ * TODO: check with SDK team for token expire specific error.
```

Comment:
Did you check with the SDK team? Do you need this TODO?

Reply:
I reached out. I think this works now, but will see if we can get a better property in the future.